### PR TITLE
tests: drivers: Fix coverity issue 198873

### LINKED ISSUE #16574: [Coverity CID :198873]Incorrect expression in /tests/drivers/uart/uart_async_api/src/test_uart_async.c
Static code scan issues seen in File: /tests/drivers/uart/uart_async_api/src/test_uart_async.c
 Category: Incorrect expression
 Function: test_chained_read
 Component: Tests
 CID: 198873
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by Mierunski**
Fix coverity issue 198873

Signed-off-by: Mieszko Mierunski <mieszko.mierunski@nordicsemi.no>~

Fixes #16574

I don't know how to check if coverity issues are fixed locally, if there is some test that I should perform please let me know

### CODE REVIEW

File: tests/drivers/uart/uart_async_api/src/test_uart_async.c [hunk 1]
```diff
@@ -132,7 +132,9 @@ void test_chained_read(void)
 		uart_tx(uart_dev, tx_buf, sizeof(tx_buf), 100);
 		zassert_equal(k_sem_take(&tx_done, 100), 0, "TX_DONE timeout");
 		zassert_equal(k_sem_take(&rx_rdy, 1000), 0, "RX_RDY timeout");
-		zassert_equal(read_len, sizeof(tx_buf),
+		size_t read_len_temp = read_len;
```
Comment (ioannisg) [inline tests/drivers/uart/uart_async_api/src/test_uart_async.c]: Just curious; why didn't you just make read_len non-volatile?

Comment (Mierunski) [reply tests/drivers/uart/uart_async_api/src/test_uart_async.c]: read_len is a global variable checked in loop and changed in interrupt. Is there a way to make it work without using volatile with enabled optimization?

Comment (ioannisg) [reply tests/drivers/uart/uart_async_api/src/test_uart_async.c]: Nope, I think this should be the right solution

**Review by ioannisg — APPROVED** [review]

**Comment by ioannisg** [comment]
@Mierunski please, rebase this one

**Pull request merged**